sql/core/src/test/resources/sql-tests/results/typeCoercion/native/concat.sql.out

```diff
@@ -40,16 +40,16 @@ FROM (
 -- !query schema
 struct<col:string>
 -- !query output
-prefix_0120.0
-prefix_1231.0
-prefix_2342.0
-prefix_3453.0
-prefix_4564.0
-prefix_5675.0
-prefix_6786.0
-prefix_7897.0
-prefix_89108.0
-prefix_910119.0
+prefix_0120
+prefix_1231
+prefix_2342
+prefix_3453
+prefix_4564
+prefix_5675
+prefix_6786
+prefix_7897
+prefix_89108
+prefix_910119
```

Comment:
@cloud-fan Do you think it is safe?

Reply:
What causes this change? Maybe `Cast.canUpCast` is not strict enough.

Reply:
It seems it is not safe when casting `IntegralType` to `DoubleType`.
```
spark.sql("create table t1(id long) using parquet")
spark.sql("insert into t1 values(1)")
spark.sql("insert into t1 values(92233720368547758L)")
spark.sql("select id, cast(cast(id as double) as string) from t1").show(false)
```
Before this pr:
```
+-----------------+----------------------------------+
|id               |CAST(CAST(id AS DOUBLE) AS STRING)|
+-----------------+----------------------------------+
|92233720368547758|9.223372036854776E16              |
|1                |1.0                               |
+-----------------+----------------------------------+
```

After this pr:
```
+-----------------+----------------------------------+
|id               |CAST(CAST(id AS DOUBLE) AS STRING)|
+-----------------+----------------------------------+
|1                |1                                 |
|92233720368547758|92233720368547758                 |
+-----------------+----------------------------------+
```

Reply:
yea, long -> double is not a 100% safe upcast because it may truncate. How about we define a `isWiderCast` in this rule and only put 100% safe upcast there?